WordPress/Classes/Services/PostService.swift

```diff
@@ -0,0 +1,673 @@
+                                    }
+                }
+            )
+        }
+    }
+
+    /**
+     Sync an initial batch of posts from the specified blog.
+
+     - note: Please note that success and/or failure are called in the context of the
+     NSManagedObjectContext supplied when the PostService was initialized, and may not
+     run on the main thread.
+
+     - parameters:
+         - postType: The type (post or page) of post to sync
+         - blog: The blog that has the posts.
+         - success: åA success block
+         - failure: A failure block
+     */
+    func sync(type: PostType, blog: Blog, success: SuccessPosts, failure: FailureBasic) {
+        self.sync(type, blog: blog, options: nil, success: success, failure: failure)
+    }
+
+    /**
+     Sync a batch of posts with the specified options from the specified blog.
+
+     - note: Please note that success and/or failure are called in the context of the
+     NSManagedObjectContext supplied when the PostService was initialized, and may not
+     run on the main thread.
+
+     - parameters:
+         - postType: The type (post or page) of post to sync
+         - options: Sync options for specific request parameters.
+         - blog: The blog that has the posts.
+         - success: A success block
+         - failure: A failure block
+    */
+    func sync(type: PostType, blog: Blog, options: PostServiceSyncOptions?, success: SuccessPosts?, failure: FailureBasic?) {
+        if let remote = self.remote(for:blog) {
+            let blogID = blog.objectID
+            let remoteOptions = self.remoteSyncParameters(for:remote, with:options)
+            remote.getPostsOfType(type.rawValue, options: remoteOptions, success: { (remotePosts) in
+                    self.managedObjectContext.performBlock({
+                        do {
+                            if let blogInContext = try self.managedObjectContext.existingObjectWithID(blogID) as? Blog {
+                                self.merge(posts: remotePosts,
+                                    type: type,
+                                    statuses: options?.statuses,
+                                    author: options?.authorID,
+                                    blog: blogInContext,
+                                    purge: options?.purgesLocalSync,
+                                    completion: { posts in
+                                        if let success = success {
+                                            success(posts)
+                                        }
+                                })
+                            }
+                        } catch let error as NSError? {
+                            DDLogSwift.logError(String("Could not retrieve blog in context with error: %@", error))
+                        }
+                    })
+                }, failure: { (error) in
+                    self.managedObjectContext.performBlock({
```

Comment:
When a closure is the last method's parameter, you can do this:

```
self.managedObjectContext.performBlock() {
   failure?(error)
}
```

(It can live as a "tail", no need to actually send it over as a parameter)